interface/src/app/analyze/analysis/sampleBin.js

```diff
@@ -77,6 +80,26 @@ function($log, $cacheFactory, $q, Sample, Activity) {
       }
     },
 
+    getSampleGroups: function() {
+      var keys = Object.keys(this.sampleGroups);
+      var samplesByGroup = {};
+      var i;
+
+      // each distinct value in sampleGroups becomes a key in samplesByGroup,
+      // and the keys of sampleGroups are collected in a list within each
+      // corresponding value of samplesByGroup
+      for (i = 0; i < keys.length; i++) {
+        if (samplesByGroup[this.sampleGroups[+keys[i]]]) {
```

Comment:
I'll try that and see if it looks better. I'm usually more bothered by too many temp vars than extra brackets, but I can see how that might help in this case. Might even get a minuscule performance boost, too, but I'm no expert in JavaScript parsing optimizations, so I could be wrong there.

- [ ] try @rzelayafavila's temp var suggestion